pkg/gadgets/snapshot/process/types/process-collector.go

```diff
@@ -36,3 +41,96 @@ type Event struct {
 func GetColumns() *columns.Columns[Event] {
 	return columns.MustCreateColumns[Event]()
 }
+
+type processTree struct {
+	process  *Event
+	children []*processTree
+}
+
+func createTree(processes []*Event) (*processTree, error) {
+	var root *processTree
+
+	nodes := make(map[int]*processTree, len(processes))
+	// Create a node for each process.
+	for _, process := range processes {
+		nodes[process.Pid] = &processTree{
+			process:  process,
+			children: make([]*processTree, 0),
+		}
+	}
+
+	// Link all nodes together.
+	for _, node := range nodes {
+		ppid := node.process.ParentPid
+		if _, ok := nodes[ppid]; !ok {
+			if root != nil {
+				return nil, fmt.Errorf("tree has two root processes: %v and %v", root, node)
+			}
+
+			root = node
+
+			continue
+		}
+
+		nodes[ppid].children = append(nodes[ppid].children, node)
+	}
+
+	if root == nil {
+		// Even if there are orphan process, they should have a parent process
+		// as they will get the reaper as parent process:
+		// https://elixir.bootlin.com/linux/v6.1.3/source/kernel/exit.c#L653
+		// Note that above code is both called when calling exit() system call
+		// and when receiving a fatal signal.
+		return nil, fmt.Errorf("container has no root process")
+	}
+
+	return root, nil
+}
+
+func treeToString(node *processTree) string {
```

Comment:
Define this as `func (t *processTree) String()` to implement the [Stringer](https://go.dev/tour/methods/17) interface?